src/discover.ts

```diff
@@ -0,0 +1,39 @@
+import * as fs from "fs";
+import * as path from "node:path";
+import { TileDatabase } from "./db/tileDatabase";
+import { TileDataset } from "./types/app";
+import { Dict } from "./types/utils";
+
+export const discoverTileDatasets = (root: string): Dict<TileDataset> => {
+    // We expect to find tile databases in a single nesting of folders under root, where each folder has the dataset
+    // name, and each *.mbtiles files in the folder has the level name. These are the dataset and level identifiers
+    // which should be used in urls to access the tile data.
+    const folders = fs
+        .readdirSync(root, { withFileTypes: true })
+        .filter((entry) => entry.isDirectory())
+        .map((entry) => entry.name);
+
+    if (folders.length) {
+        console.log("Found tile datasets:");
+    } else {
+        console.warn("No tile datasets found!");
```

Comment:
Should this a more alerting type of failure, e.g. throw and prevent app start-up because it won't be a valid instance without any db?

Reply:
Debatable I think. There's a metadata endpoint coming soon, which would reveal to a client that there aren't any datasets, and perhaps that would be more useful than just refusing to startup. Also, we are going to be adding non-tile metadata at some point, so I suppose it could be valid to have an instance with no tile datasets.